pkg/storegateway/bucket.go

```diff
@@ -1144,6 +1151,9 @@ func (s *BucketStore) getSeriesIteratorFromBlocks(
 
 	stats.update(func(stats *queryStats) {
 		stats.blocksQueried = len(batches)
```

Comment:
Does it matter that `blocksQueried` is being set here while we're `+=` the source and level counts? I don't _think_ it does but I'm not sure

Reply:
Yea, it shouldn't matter unless the `stats` object is being reused across function calls, which it shouldn't be. But I was defensive here and went with += just in case.

Reply:
Nick raised a very good point. For the streaming case, the same stats object is updated twice but the blocks are the same. It means that `blocksQueried` is updated correctly but `blocksQueriedBySourceAndLevel` is double counted.

Reply:
Thanks man!